File: go/facade/notification.go
package facade

import "fmt"

type Notification struct{}

func NewNotification() *Notification {
	return &Notification{}
}

func (n Notification) SendEmail(message string) {
	fmt.Println("메일 전송 : " + message)
}

func (n Notification) SendAppPush(message string) {
	fmt.Println("앱 푸쉬 : " + message)
}

File: java/src/main/java/facade/notification/Notification.java
package facade.notification;

public class Notification {
    public void sendEmail(String message) {
        System.out.println("메일 전송 : " + message);
    }

    public void sendAppPush(String message) {
        System.out.println("앱 푸쉬 : " + message);
    }
}
